scripts/services.js

```diff
@@ -1673,29 +1673,62 @@ var trackerCaptureServices = angular.module('trackerCaptureServices', ['ngResour
-            columns.push({id: 'created', displayName: $translate.instant('registration_date'), valueType: 'DATE', displayInListNoProgram: false, attribute: false});
-            columns.push({id: 'inactive', displayName: $translate.instant('inactive'), valueType: 'BOOLEAN', displayInListNoProgram: false, attribute: false});
-            columns = columns.concat(returnAttributes ? returnAttributes : []);
-            
-            //generate grid column for the selected program/attributes
-            angular.forEach(columns, function(column){                
-                column.attribute = angular.isUndefined(column.attribute) ? true : false;
-                column.show = false;
-
-                if( (column.id === 'orgUnitName' && ouMode !== 'SELECTED') ||
-                    column.displayInListNoProgram || 
-                    column.displayInList){
-                    column.show = true;
-                }                
-                column.showFilter = false;                
-                filterTypes[column.id] = column.valueType;
-                if(column.valueType === 'DATE' || column.valueType === 'NUMBER' ){
-                    filterText[column.id]= {};
+
+            if ( !existedColumns ) {
+                //also add extra columns which are not part of attributes (orgunit for example)
+                columns.push({id: 'orgUnitName', displayName: $translate.instant('registering_unit'), valueType: 'TEXT', displayInListNoProgram: false, attribute: false});
+                columns.push({id: 'created', displayName: $translate.instant('registration_date'), valueType: 'DATE', displayInListNoProgram: false, attribute: false});
+                columns.push({id: 'inactive', displayName: $translate.instant('inactive'), valueType: 'BOOLEAN', displayInListNoProgram: false, attribute: false});
+                columns = columns.concat(returnAttributes ? returnAttributes : []);
+                //generate grid column for the selected program/attributes
+                angular.forEach(columns, function(column)
+                {
+                    column.attribute = angular.isUndefined(column.attribute) ? true : false;
+                    column.show = false;
+
+                    if( (column.id === 'orgUnitName' && ouMode !== 'SELECTED') || column.displayInListNoProgram || column.displayInList )
+                    {
+                        column.show = true
+                    }
+                    column.showFilter = false;
+                    filterTypes[column.id] = column.valueType;
+                    if(column.valueType === 'DATE' || column.valueType === 'NUMBER' )
+                    {
+                        filterText[column.id]= {};
+                    }
+
+                });
+            }
+            else
+            {
+                // columns = angular.copy(existedColumns);
+                for ( var i = 0; i <  returnAttributes.length; i ++ )

```

Comment:
Got a error here in testing - should probably add a nullcheck or make sure the param attributes is never undefined.

Error: null is not an object (evaluating 'returnAttributes.length')
generateGridColumns@http://localhost:8081/app-00bfb5455c795016b0a7.js:6731:54